lib/Runtime/Math/WasmMath.inl

```diff
@@ -6,9 +6,115 @@
 
 namespace Wasm
 {
-/* static */
-inline int
-WasmMath::Ctz(int value)
+
+template<typename T>
+inline T WasmMath::Div( T aLeft, T aRight )
+{
+    // Todo:: Trap on aRight == 0 for int64 and uint64
+    return aLeft / aRight;
+}
+
+constexpr uint64 specialDivLeftValue = (uint64)1 << 63;
+
+template<>
+inline int64 WasmMath::Rem( int64 aLeft, int64 aRight )
+{
+    // Todo:: Trap on aRight == 0
+    return (aLeft == specialDivLeftValue && aRight == -1) ? 0 : aLeft % aRight;
+}
+
+template<>
+inline uint64 WasmMath::Rem( uint64 aLeft, uint64 aRight )
+{
+    // Todo:: Trap on aRight == 0
+    return (aLeft == specialDivLeftValue && aRight == -1) ? specialDivLeftValue : aLeft % aRight;
+}
+
+
+template<typename T> 
+inline T WasmMath::Shl( T aLeft, T aRight )
+{
+    return aLeft << aRight;
+}
+template<typename T> 
+inline T WasmMath::Shr( T aLeft, T aRight )
+{
+    return aLeft >> aRight;
+}
+
+template<typename T> 
+inline T WasmMath::ShrU( T aLeft, T aRight )
+{
+    return aLeft >> aRight;
+}
+
+#ifdef _M_IX86
+template<>
+inline int64 WasmMath::Shl( int64 aLeft, int64 aRight )
+{
+    // on x86 the compiler uses _allshl which returns 0 if aRight >= 64
```

Comment:
wasm shifts truncate to number of valid bits to shift, so instead of this can we do `aLeft << (aRight & (sizeof(T)*8-1))`? This should be shared with int32 shift, and it should also be applied to the other shifts as well